### DISCUSSION TITLE
ssc reset

### DISCUSSION THREAD

**Discussion by ghost** [Pryaxis/TShock#2294] [Q&A]
How can i reset everyone's characters without deleting  the accounts and groups?

**Comment by Kojirremer** [comment]
If you mean everyone _at once_, the only way I can think of is to manually reset their player info within ``tshock.sqlite`` by replacing all of the tsCharacter table entries' columns **except for the ``Account`` one** with starting value entries.

Be warned though, databases are fragile things. If you don't know what you are doing and just steamroll your way thinking "I'll fix it if I mess up", then you are likely going to lose the whole database, not just the inventories. Always back up your important server files (config files, databases, world file) before trying potentially unsafe things.

**Reply by ghost** [reply]
i know there is a plugin ssc reset but bruh tshock made update and now it doesnt work
but the sad thing is that tshock doesnt have the option to reset ssc players

**Reply by moisterrific** [reply]
Yes it would be nice if more SSC management commands were built-in.

**Comment by moisterrific** [comment]
Easiest way: use a plugin that lets you reset SSC stuff.

Also easy but more tedious: download SQLite Browser, open tshock.sqlite, and delete the entire `tsCharacter` table. 

There’s a minor bug with the latter method that I don’t remember if it happens too for plugins. Players end up keeping the extra slot they unlocked using Demon Heart in Expert or Master modes.

**Comment by moisterrific** [comment]
i updated it for the latest tsock version here u go

https://github.com/moisterrific/CharacterReset/releases/tag/v1.0.3
